devicetypes/smartthings/zwave-door-temp-sensor.src/zwave-door-temp-sensor.groovy

```diff
@@ -0,0 +1,299 @@
+    sensorValueEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+    sensorValueEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv1.SensorBinaryReport cmd) {
+    sensorValueEvent(cmd.sensorValue)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensoralarmv1.SensorAlarmReport cmd) {
+    sensorValueEvent(cmd.sensorState)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+    def result = []
+    def cmds = []
+
+    if (cmd.notificationType == 0x06 && cmd.event == 0x16) {
+        result << sensorValueEvent(1)
+    } else if (cmd.notificationType == 0x06 && cmd.event == 0x17) {
+        result << sensorValueEvent(0)
+    } else if (cmd.notificationType == 0x07) {
+        if (cmd.event == 0x00) {
+            result << createEvent(descriptionText: "$device.displayName covering was restored", isStateChange: true)
+            cmds = [zwave.batteryV1.batteryGet(), zwave.wakeUpV1.wakeUpNoMoreInformation()]
+            result << response(commands(cmds, 1000))
+        } else if (cmd.event == 0x01 || cmd.event == 0x02) {
+            result << sensorValueEvent(1)
+        } else if (cmd.event == 0x03) {
+            result << createEvent(descriptionText: "$device.displayName covering was removed", isStateChange: true)
+            cmds = [zwave.manufacturerSpecificV2.manufacturerSpecificGet(), zwave.wakeUpV1.wakeUpNoMoreInformation()]
+            result << response(commands(cmds, 1000))
+        }
+    } else if (cmd.notificationType) {
+        def text = "Notification $cmd.notificationType: event ${([cmd.event] + cmd.eventParameter).join(", ")}"
+        result << createEvent(name: "notification$cmd.notificationType", value: "$cmd.event", descriptionText: text, displayed: false)
+    } else {
+        def value = cmd.v1AlarmLevel == 255 ? "active" : cmd.v1AlarmLevel ?: "inactive"
+        result << createEvent(name: "alarm $cmd.v1AlarmType", value: value, displayed: false)
+    }
+
+    result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+    def event = createEvent(descriptionText: "${device.displayName} woke up", isStateChange: false)
+    def cmds = []
+
+    // If any of our initial request didn't make it, request the sensor data again
+    if (!state.MSR) {
+        cmds << zwave.manufacturerSpecificV2.manufacturerSpecificGet()
+    }
+
+    if (device.currentValue("contact") == null) {
+        cmds << zwave.sensorBinaryV2.sensorBinaryGet(sensorType: zwave.sensorBinaryV2.SENSOR_TYPE_DOOR_WINDOW)
+    }
+
+    if ((!state.lastbat) || ((now() - state.lastbat) > (53*60*60*1000))) {
+        cmds << zwave.batteryV1.batteryGet()
+    }
+
+    cmds << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 1)
+    cmds << zwave.wakeUpV1.wakeUpNoMoreInformation()
+
+    [event, response(commands(cmds, 1000))]
```

Comment:
I would remove the delays if this was up to me, maybe there is a reason for these delays that should be documented.

Reply:
I was following the documentation for [building-z-wave-device-handlers](https://docs.smartthings.com/en/latest/device-type-developers-guide/building-z-wave-device-handlers.html):

>If you want to send more than one Z-Wave command, you can return a list of formatted command strings. It is often a good idea to add a delay between commands to give the device an opportunity to finish processing each command and possibly send a response before receiving the next command. To add a delay between commands, include a string of the form "delay N" where N is the number of milliseconds to delay. There is a helper method delayBetween() that will take a list of commands and insert delay commands between them:

Reply:
If the hub is queuing the commands anyways, seems like delay between commands should be handled by the hub too.

Reply:
I do not disagree with you.

Reply:
Unless someone disagrees, I'm going to leave the delays alone for now since they seem to work. We communicated with Inovelli regarding the delays and found out their door/temp sensor has been pulled from production for the time being due to some unreliable behavior.